storage/cloudspanner/map_storage.go

```diff
@@ -269,14 +273,104 @@ func (tx *mapTX) Set(ctx context.Context, index []byte, value *trillian.MapLeaf)
 	return stx.BufferWrite([]*spanner.Mutation{m})
 }
 
-// GetTiles is not implemented.
+// GetTiles reads the Merkle tree tiles with the given root IDs at the given
+// revision. A tile is empty if it is missing from the returned slice.
 func (tx *mapTX) GetTiles(ctx context.Context, rev int64, ids []tree.NodeID2) ([]smt.Tile, error) {
-	return nil, errors.New("not implemented")
+	rootIDs := make([]tree.NodeID, 0, len(ids))
+	for _, id := range ids {
+		rootIDs = append(rootIDs, tree.NewNodeIDFromID2(id))
+	}
+
+	f, err := tx.treeTX.getTilesFunc(ctx, rev)
+	if err != nil {
+		return nil, err
+	}
+	subs, err := f(rootIDs)
```

Comment:
For me  `subs` was not immediately obvious. Substitutes was my initial expectation. Because the fn that these are returned from is a little indirect, finding the documentation on what they are isn't clear.

My preference would be to explicitly state their type here, but renaming to `subtrees` or `protos` might also help.

Reply:
done